### ISSUE TITLE
Scylla start took more time then expected - 10 min (CDC test)

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylla#8297]
*Installation details*
Scylla version (or git commit hash): `4.4.rc3-0.20210304.c2d924757 with build-id 1d0929cbdc3fd39b79157e974cd8d08cd2610afb`
Cluster size: 6 nodes (i3.4xlarge)
OS (RHEL/CentOS/Ubuntu/AWS AMI): `ami-0242812f1bd719d03` (aws: eu-west-1)

Test: `longevity-cdc-3d-400gb-test`
Test name: `longevity_test.LongevityTest.test_custom_time`
Test config file(s):
- [longevity-cdc-3d-400gb.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/440fdb6ae5c22e187f127314a7f37f1488c6cc18/test-cases/longevity/longevity-cdc-3d-400gb.yaml)

**Issue description**

Scylla start took more time then expected - 10 min

Nemesis `StopWaitStartService` (HH is default):
1. stop Scylla service 
2. wait 5 min
3. start Scylla service

This is the flow of service start:
```
Mar 11 11:59:24 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 systemd[1]: Starting Scylla Server...
Mar 11 11:59:25 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]: Scylla version 4.4.rc3-0.20210304.c2d924757 with build-id 1d0929cbdc3fd39b79157e974cd8d08cd2610afb starting ...
Mar 11 11:59:27 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] init - Scylla API server listening on 127.0.0.1:10000 ...
Mar 11 12:08:55 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 systemd[1]: Started Scylla Server.
Mar 11 12:08:55 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 systemd[1]: Started Run Scylla Housekeeping daily mode.
Mar 11 12:08:55 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 systemd[1]: Started Scylla JMX.
Mar 11 12:08:55 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 systemd[1]: Started Run Scylla Housekeeping restart mode.
Mar 11 12:09:02 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 systemd[1]: Started Scylla Housekeeping restart mode.
```

Then compactions were started, got `reader_concurrency_semaphore`:
```
Mar 11 12:09:02 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 9] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:
                                                                             Permits with state admitted, sorted by memory
                                                                             memory        count        name
                                                                             2842K        1        cdc_test.test_table_postimage:mutation-query
                                                                             5429K        7        cdc_test.test_table_preimage_postimage:data-query
                                                                             16M        14        cdc_test.test_table:data-query
                                                                             85M        24        cdc_test.test_table_postimage:data-query
                                                                             104M        31        cdc_test.test_table_preimage:data-query
                                                                             
                                                                             214M        77        total
                                                                             
                                                                             Permits with state waiting, sorted by count
                                                                             count        memory        name
                                                                             1        0B        cdc_test.test_table_preimage:mutation-query
                                                                             3        0B        cdc_test.test_table_preimage_postimage:mutation-query
                                                                             23        17K        cdc_test.test_table_postimage:mutation-query
                                                                             103        16K        cdc_test.test_table_preimage_postimage:data-query
                                                                             116        36K        cdc_test.test_table:data-query
                                                                             198        9K        cdc_test.test_table_postimage:data-query
                                                                             263        12K        cdc_test.test_table_preimage:data-query
                                                                             
                                                                             707        90K        total
                                                                             
                                                                             Permits with state registered, sorted by count
                                                                             count        memory        name
                                                                             
                                                                             0        0B        total
                                                                             
                                                                             Total: permits: 784, memory: 214M

Mar 11 12:09:03 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 2] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:
                                                                             Permits with state admitted, sorted by memory
                                                                             memory        count        name
                                                                             5980K        2        cdc_test.test_table_preimage:mutation-query
                                                                             17M        17        cdc_test.test_table_preimage_postimage:data-query
                                                                             27M        21        cdc_test.test_table:data-query
                                                                             46M        15        cdc_test.test_table_preimage:data-query
                                                                             75M        26        cdc_test.test_table_postimage:data-query
                                                                             
                                                                             172M        81        total
                                                                             
                                                                             Permits with state waiting, sorted by count
                                                                             count        memory        name
                                                                             1        0B        cdc_test.test_table:mutation-query
                                                                             3        0B        cdc_test.test_table_preimage_postimage:mutation-query
                                                                             4        0B        cdc_test.test_table_preimage:mutation-query
                                                                             27        26K        cdc_test.test_table_postimage:mutation-query
                                                                             84        9K        cdc_test.test_table_preimage_postimage:data-query
                                                                             111        31K        cdc_test.test_table:data-query
                                                                             148        10K        cdc_test.test_table_postimage:data-query
                                                                             190        15K        cdc_test.test_table_preimage:data-query
                                                                             
                                                                             568        90K        total
                                                                             
                                                                             Permits with state registered, sorted by count
                                                                             count        memory        name
                                                                             
                                                                             0        0B        total
                                                                             
                                                                             Total: permits: 649, memory: 172M
Mar 11 12:09:03 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 11] LeveledManifest - Leveled compaction strategy is restoring invariant of level 1 by compacting 2 sstables on behalf of cdc_test.test_table_preimage_postimage
..................................................
Mar 11 12:09:04 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 4] reader_concurrency_semaphore - Semaphore _read_concurrency_sem: timed out, dumping permit diagnostics:
                                                                             Permits with state admitted, sorted by memory
                                                                             memory        count        name
                                                                             3693K        1        cdc_test.test_table_postimage:mutation-query
                                                                             28M        23        cdc_test.test_table:data-query
                                                                             33M        11        cdc_test.test_table_postimage:data-query
                                                                             40M        31        cdc_test.test_table_preimage_postimage:data-query
                                                                             64M        21        cdc_test.test_table_preimage:data-query
                                                                             
                                                                             169M        87        total
                                                                             
                                                                             Permits with state waiting, sorted by count
                                                                             count        memory        name
                                                                             3        0B        cdc_test.test_table_preimage:mutation-query
                                                                             3        0B        cdc_test.test_table_postimage:mutation-query
                                                                             3        0B        cdc_test.test_table_preimage_postimage:mutation-query
                                                                             80        9K        cdc_test.test_table_postimage:data-query
                                                                             105        15K        cdc_test.test_table_preimage_postimage:data-query
                                                                             149        45K        cdc_test.test_table:data-query
                                                                             153        7K        cdc_test.test_table_preimage:data-query
                                                                             
                                                                             496        76K        total
                                                                             
                                                                             Permits with state registered, sorted by count
                                                                             count        memory        name
                                                                             
                                                                             0        0B        total
                                                                             
                                                                             Total: permits: 583, memory: 169M
```

Continue with starting:
```
Mar 11 12:09:15 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] storage_service - Node 10.0.1.186 state jump to normal

Mar 11 12:09:17 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] storage_service - NORMAL: node is now in normal status

Mar 11 12:09:17 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] init - Waiting for gossip to settle before accepting client requests...
Mar 11 12:09:17 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 4] compaction - [Compact system.local 98fbd960-8262-11eb-981a-000000000009] Compacted 2 sstables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/md-354-big-Data.db:level=0, ]. 18kB to 12kB (~66% of original) in 2313ms = 5kB/s. ~256 total partitions merged to 1.
Mar 11 12:09:19 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 8] compaction - [Compact system.peers 9a79c6d0-8262-11eb-b9d7-000000000001] Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/md-456-big-Data.db:level=0, ]. 17kB to 11kB (~65% of original) in 1109ms = 10kB/s. ~256 total partitions merged to 1.

Mar 11 12:09:32 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] cql_server_controller - Starting listening for CQL clients on 10.0.1.186:9042 (unencrypted, non-shard-aware)
Mar 11 12:09:32 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] cql_server_controller - Starting listening for CQL clients on 10.0.1.186:19042 (unencrypted, shard-aware)
Mar 11 12:09:32 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] thrift_controller - Thrift server listening on 10.0.1.186:9160 ...
Mar 11 12:09:32 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] init - serving

Mar 11 12:09:32 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] init - Scylla version 4.4.rc3-0.20210304.c2d924757 initialization completed.
```

No reshaping during the start

Restore Monitor Stack command: `$ hydra investigate show-monitor c1aa2d4f-9121-4de7-a351-29ebe67a8c0a`
Show all stored logs command: `$ hydra investigate show-logs c1aa2d4f-9121-4de7-a351-29ebe67a8c0a`

Test id: `c1aa2d4f-9121-4de7-a351-29ebe67a8c0a`

Logs:

db-cluster - [https://cloudius-jenkins-test.s3.amazonaws.com/c1aa2d4f-9121-4de7-a351-29ebe67a8c0a/20210313_204229/db-cluster-c1aa2d4f.zip](https://cloudius-jenkins-test.s3.amazonaws.com/c1aa2d4f-9121-4de7-a351-29ebe67a8c0a/20210313_204229/db-cluster-c1aa2d4f.zip)

sct-runner - [https://cloudius-jenkins-test.s3.amazonaws.com/c1aa2d4f-9121-4de7-a351-29ebe67a8c0a/20210313_204229/sct-runner-c1aa2d4f.zip](https://cloudius-jenkins-test.s3.amazonaws.com/c1aa2d4f-9121-4de7-a351-29ebe67a8c0a/20210313_204229/sct-runner-c1aa2d4f.zip)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-4.4/job/longevity/job/longevity-cdc-3d-400gb-test/2/)

**Comment by slivne** [comment]
- Lets validate the node stopped in a clean manner - if it had the commitlogs should have been cleared and the boot should have been easy - if it hadn't stopped in a clean manner - we may have replay of commitlogs - and we need to check that (especially with cdc).

**Comment by bhalevy** [comment]
According to the log file it is just reshape that takes a long time, known issue.
```
Mar 11 11:59:33 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 0] sstable_directory - Table cdc_test.test_table_preimage_postimage with compaction strategy LeveledCompactionStrategy found SSTables that need reshape. Starting reshape process
...
Mar 11 12:08:54 longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-7 scylla[45957]:  [shard 5] compaction - [Reshape cdc_test.test_table_preimage_postimage 07fdc6d0-8262-11eb-944a-00000000000b] Reshaped 14 sstables to [/var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081
bd11ebbaf800000000000b/md-20123-big-Data.db:level=0, ]. 5GB to 4GB (~86% of original) in 222610ms = 22MB/s. ~1165312 total partitions merged to 990067.
```

**Comment by bhalevy** [comment]
See
https://github.com/scylladb/scylla/issues/7829
https://github.com/scylladb/scylla/issues/7738

Cc @raphaelsc

**Comment by bhalevy** [comment]
Is LCS the right compaction strategy for the preimage_postimage table?

**Comment by avikivity** [comment]
That's probably the base table. But why did it need reshape?

**Comment by raphaelsc** [comment]
Just sent patch which changes relaxed mode (used during initialization) of LCS reshape to tolerate more sstables in L0 (up to max_threshold), which is the right thing to do as LCS regular will be able to merge those L0 sstables (< max_threshold) into L1. Somebody please review the patch.

**Comment by juliayakovlev** [comment]
@raphaelsc  Do you need this cluster or I can terminate it?

**Comment by raphaelsc** [comment]
> @raphaelsc Do you need this cluster or I can terminate it?

yes, please.

**Comment by raphaelsc** [comment]
i think this is potentially fixed by https://github.com/scylladb/scylla/commit/e53cedabb118156ecfe47ff0b4f8f5370cec08ee, as reshape will no longer be triggered on a scenario where L0 is still below max threshold. so i am closing this issue...

**Comment by slivne** [comment]
@raphaelsc is this a regression ?

**Comment by raphaelsc** [comment]
> @raphaelsc is this a regression ?

don't think so. this problem is there since reshape was introduced.

**Comment by raphaelsc** [comment]
> > @raphaelsc is this a regression ?
> 
> don't think so. this problem is there since reshape was introduced.

not a regression in reshape, but reshape itself caused a regression in the system.

**Comment by raphaelsc** [comment]
so all versions with reshape should receive this fix.

**Comment by slivne** [comment]
I don't understand why there was a reshape to begin with - this is the main concern I have:

@juliayakovlev why was there a reshape ?

I found this:

```
itions merged to 1.
< t:2021-03-10 20:42:01,224 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:52+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: [shard 7] compaction - [Compact cdc_test.test_table_scylla_cdc_log 0b94fbf0-81e1-11eb-8ed8-000000000000] Compacting [/var/lib/scylla/data/cdc_test/test_table_scylla_cdc_log-0c612ff281bd11ebaa8400000000000a/md-1015-big-Data.db:level=0, ]
< t:2021-03-10 20:42:01,224 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:53+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: [shard 7] compaction - [Compact cdc_test.test_table_scylla_cdc_log 0b94fbf0-81e1-11eb-8ed8-000000000000] Compacted 1 sstables to []. 254MB to 0 bytes (~0% of original) in 0ms = 0 bytes/s. ~768 total partitions merged to 0.
< t:2021-03-10 20:42:01,226 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:54+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: [shard 13] compaction - [Compact cdc_test.test_table c2c7e630-81e0-11eb-8624-000000000008] Compacted 4 sstables to [/var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4689-big-Data.db:level=1, ]. 474MB to 168MB (~35% of original) in 123464ms = 1MB/s. ~99584 total partitions merged to 35518.
< t:2021-03-10 20:42:01,226 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:54+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !NOTICE  | sudo: scyllaadm : TTY=unknown ; PWD=/home/scyllaadm ; USER=root ; COMMAND=/sbin/reboot
< t:2021-03-10 20:42:01,227 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:54+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: [shard 13] compaction - [Compact cdc_test.test_table 0c75bf00-81e1-11eb-8624-000000000008] Compacting [/var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4437-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-3667-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2197-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-3625-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2071-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2267-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2239-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2099-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2015-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-3751-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2113-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2435-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2127-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4451-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-3639-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2169-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2743-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2057-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2225-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2295-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4171-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2183-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-3569-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2043-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-3135-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-1987-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2141-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4465-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-2001-big-Data.db:level=2, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4689-big-Data.db:level=1, ]
< t:2021-03-10 20:42:01,232 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:54+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | crond[1406]: (CRON) INFO (Shutting down)
< t:2021-03-10 20:42:01,233 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:54+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !ERR     | systemd-logind: Failed to abandon session scope: Connection reset by peer
< t:2021-03-10 20:42:01,235 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:41:54+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | chronyd[725]: chronyd exiting
< t:2021-03-10 20:43:04,376 f:cluster.py      l:984  c:sdcm.cluster_aws     p:DEBUG > Node longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 [34.248.20.70 | 10.0.0.253] (seed: False): Failed to get uptime during reboot, Unable to run "uptime -s": failed connecting to "10.0.0.253" during 60s
< t:2021-03-10 20:44:14,386 f:cluster.py      l:984  c:sdcm.cluster_aws     p:DEBUG > Node longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 [34.248.20.70 | 10.0.0.253] (seed: False): Failed to get uptime during reboot, Unable to run "uptime -s": failed connecting to "10.0.0.253" during 60s
< t:2021-03-10 20:44:16,238 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 0T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !NOTICE  | kernel: Linux version 5.11.3-1.el7.elrepo.x86_64 (mockbuild@Build64R7) (gcc (GCC) 9.3.1 20200408 (Red Hat 9.3.1-2), GNU ld version 2.32-16.el7) #1 SMP Wed Mar 3 19:20:04 EST 2021
< t:2021-03-10 20:44:16,238 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: Command line: BOOT_IMAGE=/boot/vmlinuz-5.11.3-1.el7.elrepo.x86_64 root=UUID=9cff3d69-3769-4ad9-8460-9c54050583f9 ro console=tty0 crashkernel=auto net.ifnames=0 console=ttyS0 clocksource=tsc tsc=reliable net.ifnames=0 LANG=en_US.UTF-8
< t:2021-03-10 20:44:16,239 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: x86/fpu: Supporting XSAVE feature 0x001: 'x87 floating point registers'
< t:2021-03-10 20:44:16,239 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: x86/fpu: Supporting XSAVE feature 0x002: 'SSE registers'
< t:2021-03-10 20:44:16,239 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: x86/fpu: Supporting XSAVE feature 0x004: 'AVX registers'
< t:2021-03-10 20:44:16,240 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: x86/fpu: xstate_offset[2]:  576, xstate_sizes[2]:  256
< t:2021-03-10 20:44:16,240 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: x86/fpu: Enabled xstate features 0x7, context size is 832 bytes, using 'standard' format.
< t:2021-03-10 20:44:16,240 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: BIOS-provided physical RAM map:
< t:2021-03-10 20:44:16,240 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: BIOS-e820: [mem 0x0000000000000000-0x000000000009dfff] usable
< t:2021-03-10 20:44:16,241 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: BIOS-e820: [mem 0x000000000009e000-0x000000000009ffff] reserved
< t:2021-03-10 20:44:16,241 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: BIOS-e820: [mem 0x00000000000e0000-0x00000000000fffff] reserved
< t:2021-03-10 20:44:16,241 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: BIOS-e820: [mem 0x0000000000100000-0x000000007fffffff] usable
< t:2021-03-10 20:44:16,242 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:43:58+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | kernel: BIOS-e820: [mem 0x00000000fc000000-0x00000000ffffffff] reser:
```

So the node was restarted - did we change anything - unless it was msb bits - the answer is no
```
< t:2021-03-10 20:44:16,516 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:44:09+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: parsed command line options: [blocked-reactor-notify-ms: 500, abort-on-lsa-bad-alloc: 1, abort-on-seastar-bad-alloc, abort-on-internal-error: 1, abort-on-ebadf: 1, enable-sstable-key-validation: 1, log-to-syslog: 1, log-to-stdout: 0, default-log-level: info, network-stack: posix, io-properties-file: /etc/scylla.d/io_properties.yaml, cpuset: 1-7,9-15, lock-memory: 1]
< t:2021-03-10 20:44:16,517 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:44:10+00:00  longevity-cdc-3d-400gb-4-4
```

```
< t:2021-03-10 20:44:16,516 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:44:09+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: Scylla version 4.4.rc3-0.20210304.c2d924757 with build-id 1d0929cbdc3fd39b79157e974cd8d08cd2610afb starting ...
< t:2021-03-10 20:44:16,516 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:44:09+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: command used: "/usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --abort-on-internal-error 1 --abort-on-ebadf 1 --enable-sstable-key-validation 1 --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15 --lock-memory=1"
```

cpuset is the same

the reshaped files

```
< t:2021-03-10 20:44:31,568 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:44:26+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: [shard 1] compaction - [Reshape cdc_test.test_table 67414e90-81e1-11eb-b741-000000000002] Reshaping [/var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5419-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-4901-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5475-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5545-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5377-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5139-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5153-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5195-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5167-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5209-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5181-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5251-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5237-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5223-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5125-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5279-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5097-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5531-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5083-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5069-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5055-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table-0c612ff081bd11ebaa8400000000000a/md-5321-big-Data.db:level=0, ]
```

have been created before the reboot and where included in comapctions

```
< t:2021-03-10 20:29:10,912 f:cluster.py      l:1395 c:sdcm.cluster         p:DEBUG > 2021-03-10T20:29:10+00:00  longevity-cdc-3d-400gb-4-4-db-node-c1aa2d4f-4 !INFO    | scylla: [shard 1] compaction - [Compact cdc_test.test_table_preimage
_postimage 456f1510-81df-11eb-a245-000000000009] Compacting [/var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5391-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_posti
mage-1bc85d1081bd11ebbaf800000000000b/md-5517-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5433-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage
_postimage-1bc85d1081bd11ebbaf800000000000b/md-5447-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5419-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_pr
eimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5531-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5825-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_ta
ble_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5643-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5615-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/t
est_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5573-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5559-big-Data.db:level=0, /var/lib/scylla/data/cdc_
test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5489-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5545-big-Data.db:level=0, /var/lib/scylla/dat
a/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5685-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5713-big-Data.db:level=0, /var/lib/scyl
la/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5671-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5629-big-Data.db:level=0, /var/li
b/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5993-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5755-big-Data.db:level=0, /
var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5881-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5699-big-Data.db:leve
l=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5811-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5797-big-Data.d
b:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5727-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5741-big-
Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5769-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-578
3-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5839-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/
md-5867-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-5853-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf80000000
0000b/md-6665-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-6301-big-Data.db:level=0, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf80
0000000000b/md-7015-big-Data.db:level=1, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-6987-big-Data.db:level=1, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11e
bbaf800000000000b/md-7127-big-Data.db:level=1, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-7155-big-Data.db:level=1, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d108
1bd11ebbaf800000000000b/md-7001-big-Data.db:level=1, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc85d1081bd11ebbaf800000000000b/md-7085-big-Data.db:level=1, /var/lib/scylla/data/cdc_test/test_table_preimage_postimage-1bc
85d1081bd11ebbaf800000000000b/md-7169-big-Data.db:level=1, ]
``` 

(and I assume survived them ... :) ).

- this is a bit strange - we had a compaction with levels=0, level=1 in LCS
  - so did we promote sstables from level=0 (many) to level=1 and compacted them together
  - if in compaction we find nothing needs to be rewritten do we keep the sstable - do we rewrite the summary file with the level ?

and compaction may have not ended

So why was reshape triggered ? - if we have too many files in level=0 will this trigger a reshape ?

**Comment by raphaelsc** [comment]
> 
> So why was reshape triggered ? - if we have too many files in level=0 will this trigger a reshape ?

yes, that's why I patched LCS reshape to tolerate more sstables in level 0. if no resharding, compaction falling behind is the reason for reshape being triggered on boot..

**Comment by raphaelsc** [comment]
looks like we miss documentation to explain reshape behavior on boot. i'll write one.

**Comment by slivne** [comment]
Hm ... yet this is exactly what we want todo - we want to reshape the files on boot to make sure that when we get reads we will not have a backlog for compaction and the files will be aligned according to compaction strategy - so why push this out.

The problem with the approach of allowing more sstables in L0 is that the node is slow because its cache its clod - its slow because it doesn;t have any data in memetables it will be now supper slow because the sstables are not aligned to compaction starategy so reads will be slower.


Maybe the issue is that we timeout for systemd on boot - not that it takes long - you added a flag to bypass reshape on boot - it can be used for this - I don;t think we should relax the L0 on boot since once its up it will be super slow.

**Comment by raphaelsc** [comment]
> Hm ... yet this is exactly what we want todo - we want to reshape the files on boot to make sure that when we get reads we will not have a backlog for compaction and the files will be aligned according to compaction strategy - so why push this out.
> 
> The problem with the approach of allowing more sstables in L0 is that the node is slow because its cache its clod - its slow because it doesn;t have any data in memetables it will be now supper slow because the sstables are not aligned to compaction starategy so reads will be slower.
> 
> Maybe the issue is that we timeout for systemd on boot - not that it takes long - you added a flag to bypass reshape on boot - it can be used for this - I don;t think we should relax the L0 on boot since once its up it will be super slow.

we aren't pushing it out. reshape on boot runs on relaxed mode (please don't interpret relaxed as too relaxed or not completely conforming to the strategy criteria), allowing only min_threshold in L0 (4) is considered strict, with relaxed mode, like in STCS, we can allow up to max_threshold. also, when node is online, those L0 sstables can be directly merged into L1 instead of being compacted in place, which reduces WA.

**Comment by slivne** [comment]
Ok, its clear I am not the expert - yet I'll try to rephrase to align to what we think we should have.

A node that reboots and doesn't get any writes/hints should endup ready to serve reads without having any sort of backlog for compaction. 
- so if the relaxed mode is switched off at the end of the bootstrap and the node will have to compact because it moved from "relaxed" to "strict" we will need to fix that.
- there is a difference between a node bootstrapping and a node that is rebooting. 

- A node that is bootstrapping maybe more relaed while its receiging the data - but at the end before moving to Up NORMAL should move to strict and clear all backlog
- A node that is restarting should not be relaxed at all in my view.

**Comment by raphaelsc** [comment]
> Ok, its clear I am not the expert - yet I'll try to rephrase to align to what we think we should have.
> 
> A node that reboots and doesn't get any writes/hints should endup ready to serve reads without having any sort of backlog for compaction.
> 
> * so if the relaxed mode is switched off at the end of the bootstrap and the node will have to compact because it moved from "relaxed" to "strict" we will need to fix that.
> * there is a difference between a node bootstrapping and a node that is rebooting.
> * A node that is bootstrapping maybe more relaxed while its receiving the data (writes + streaming) - but at the end before moving to Up NORMAL should move to strict and clear all backlog

That's what we do today without repair-based operations, by temporarily increasing min threshold and restoring it to previous value on completion. But backlog is only eventually cleared through regular compactions.

With repair-based operations, we're now reshaping the new sstables, on operation completion, in maintenance mode. That's possible because the new sstables are disjoint and can be read from efficiently by taking advantage of their disjointness. 

> * A node that is restarting should not be relaxed at all in my view.

We decided to be more tolerant on restart mainly because of boot time. For example, if I stop my node with compactions going on, should scylla start reshape and wait for compactions which were previously stopped? Input from field is that a significant increase in boot time isn't very acceptable.

Please note that relaxed mode has a higher tolerance than strict, but will not allow tables to be badly shaped (meaning bad read performance) once node is online. 

With relaxed mode on LCS, for example, it's still guaranteed that all levels are disjoint, but L0 can have up to max_threshold sstables, which can be later directly merged into L1. Compacting L0 sstables in L0 itself to be later merged, once node is online, into L1 is very inefficient in terms of WA.

Reshape on boot shouldn't be seen as a solution for reducing the backlog to 0, but rather to fix more serious issues like, LCS: fixing overlapping ssts in levels > 0 and exceeding # of sstables in L0, TWCS: sstables spanning more than 1 window, STCS: exceeding # of sstables in a given size tier. If compaction is really falling behind, we have a problem in the controller or/and saturation, and so even if reshape on boot reduces backlog to 0, compaction will eventually fall behind again, anyway.

**Comment by slivne** [comment]
If I restart the node in a clean manner - the ndoe should start fast

If I restart the node in an non clean manner like here - simply restart the machine and don't shutdown scylla in a clean manner - it will be slow - replay commitlogs / reshape - thats fine. 
- I can bypass the reshape with the new flag you added if I want to bootstrap it in a fast mode regardless of the fact that I made a mess with a non clean shutdown - I should not have the restarted node slower because it has to reshape in parallel to having its cache cold.

A slow node will cause all the cluster to be slower - its not ok to assume this is fine - there is a real penalty with making the bootstrap fast reshapoe was created to fix that and assure the node starting will be in a "good shape" and will not cause the whole cluster to be wloer.

**Comment by raphaelsc** [comment]
> If I restart the node in a clean manner - the ndoe should start fast
> 
> If I restart the node in an non clean manner like here - simply restart the machine and don't shutdown scylla in a clean manner - it will be slow - replay commitlogs / reshape - thats fine.

If I stopped Scylla now, it's expected that sstables are in good shape unless my node is saturated or there's a problem in the controller. 

Previous to shutdown, let's say backlog is X, where X > 0. Before shutdown, there can be compaction work trying to reduce backlog to < X while node was online. Why should reshape be started to reduce backlog to below X if I could just start the node and have the exact same scenario as before? Which is: backlog X with ongoing compactions working to reduce backlog to below X. 

> 
> * I can bypass the reshape with the new flag you added if I want to bootstrap it in a fast mode regardless of the fact that I made a mess with a non clean shutdown - I should not have the restarted node slower because it has to reshape in parallel to having its cache cold.

Avi disliked the option to bypass reshape on boot. Instead, we'll provide a way to abort the operation while it's in flight. That's open to discussion though.

> 
> A slow node will cause all the cluster to be slower - its not ok to assume this is fine - there is a real penalty with making the bootstrap fast reshapoe was created to fix that and assure the node starting will be in a "good shape" and will not cause the whole cluster to be wloer.

Perhaps, you're interpreting relaxed mode as allowing the node to be in a bad shape, but I don't think that's true. Perhaps the term relaxed mode was badly chosen. With relaxed mode, I allow the ongoing compactions which were running or scheduled to run, before shutdown, to re-start after node is online. Relaxed mode will still work to leave table in a good shape if not so. Backlog != 0 doesn't mean bad shape. High backlog means bad shape.

Bootstrap can be made fast with repair-based operation, because new sstables are disjoint (there's only one sstable per vnode range). So we can reshape them in maintenance mode while serving them from a maintenance set which takes advantage of their disjointness.

**Comment by raphaelsc** [comment]
By the way, for bootstrap, we're using strict mode, not relaxed. For regular boot, we're using relaxed.

**Comment by bhalevy** [comment]
> A slow node will cause all the cluster to be slower - its not ok to assume this is fine - there is a real penalty with making the bootstrap fast reshapoe was created to fix that and assure the node starting will be in a "good shape" and will not cause the whole cluster to be wloer.

I'm not sure I agree to that.  It's equivalent to choose to restart during normal operation mode if compaction is running behind.
According to your argument increased read-amplification causes the node to be slower and with it the whole cluster, so it would be better for the node to restart itself and run reshape compaction to catch up.  I don't think it holds water since while being off-line writers accumulate and then they need to be replayed with HH or repaired and compaction will need to run on these writes, so it essentially accumulates a back-log on other nodes, while it's not serving writes.

We need to fix this case by fixing the steady state, not by working around it after restart.

**Comment by k0machi** [comment]
Had a similiar issue on some nodes running `longevity-twcs-48h-test#25`, nodes would take around 5-15 minutes to reshape data on bootup, causing certain timeouts to be reached.

## Installation details
Kernel Version: 5.13.0-1019-aws
Scylla version (or git commit hash): `5.1.dev-20220317.c45050895403` with build-id `076f513b6143670def988c7626389b270411f8f7`
Cluster size: 4 nodes (i3en.2xlarge)
Scylla Nodes used in this run:
- longevity-twcs-48h-master-db-node-2acb1d5c-8 (54.195.204.64 | 10.0.3.165) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-7 (63.32.45.143 | 10.0.0.235) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-6 (54.216.174.11 | 10.0.2.196) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-5 (54.170.137.53 | 10.0.0.106) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-4 (3.250.227.110 | 10.0.0.15) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-3 (34.248.103.231 | 10.0.2.143) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-2 (54.154.214.184 | 10.0.1.237) (shards: 8)
- longevity-twcs-48h-master-db-node-2acb1d5c-1 (34.249.11.226 | 10.0.1.42) (shards: 8)
- 
OS / Image: `ami-0d27db767997d0ec4` (aws: eu-west-1)

Test: `longevity-twcs-48h-test`
Test id: `2acb1d5c-3e77-4abd-93e7-ae1d3fa89903`
Test name: `longevity/longevity-twcs-48h-test`
Test config file(s):

- [longevity-twcs-48h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/1cf1a7533b0af12a4484921aa18db289b82b4f94/test-cases/longevity/longevity-twcs-48h.yaml)

- Restore Monitor Stack command: `$ hydra investigate show-monitor 2acb1d5c-3e77-4abd-93e7-ae1d3fa89903`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=2acb1d5c-3e77-4abd-93e7-ae1d3fa89903)
- Show all stored logs command: `$ hydra investigate show-logs 2acb1d5c-3e77-4abd-93e7-ae1d3fa89903`

## Logs:
- **db-cluster-2acb1d5c.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/db-cluster-2acb1d5c.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/db-cluster-2acb1d5c.tar.gz)
- **monitor-set-2acb1d5c.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/monitor-set-2acb1d5c.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/monitor-set-2acb1d5c.tar.gz)
- **loader-set-2acb1d5c.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/loader-set-2acb1d5c.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/loader-set-2acb1d5c.tar.gz)
- **sct-runner-2acb1d5c.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/sct-runner-2acb1d5c.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/2acb1d5c-3e77-4abd-93e7-ae1d3fa89903/20220326_205719/sct-runner-2acb1d5c.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-twcs-48h-test/25/)

**Comment by KnifeyMoloko** [comment]
Also reproduced with below run. Startup took 10 minutes:


## Installation details

Kernel Version: 5.13.0-1022-aws
Scylla version (or git commit hash): `5.1.dev-20220428.333fdcb3f544` with build-id `415dea72992f5c04c11c4171a5b878d2e1c48fa9`
Cluster size: 4 nodes (i3en.2xlarge)

Scylla Nodes used in this run:
- longevity-twcs-48h-master-db-node-98f6aa26-8 (34.247.88.105 | 10.0.0.204) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-7 (34.240.81.69 | 10.0.2.167) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-6 (34.245.59.217 | 10.0.3.211) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-5 (54.216.67.78 | 10.0.1.142) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-4 (52.50.228.214 | 10.0.0.214) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-3 (54.217.58.245 | 10.0.2.88) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-2 (52.30.37.46 | 10.0.2.138) (shards: 8)
- longevity-twcs-48h-master-db-node-98f6aa26-1 (52.48.117.9 | 10.0.3.168) (shards: 8)


OS / Image: `ami-0bad52798af1cde66` (aws: eu-west-1)

Test: `longevity-twcs-48h-test`
Test id: `98f6aa26-f45b-4808-8aec-e3cf277e33fe`
Test name: `scylla-master/longevity/longevity-twcs-48h-test`
Test config file(s):

- [longevity-twcs-48h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/3615a6c5d451a30d2929dd1593784ba7c643783f/test-cases/longevity/longevity-twcs-48h.yaml)

## Issue description

- Restore Monitor Stack command: `$ hydra investigate show-monitor 98f6aa26-f45b-4808-8aec-e3cf277e33fe`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=98f6aa26-f45b-4808-8aec-e3cf277e33fe)
- Show all stored logs command: `$ hydra investigate show-logs 98f6aa26-f45b-4808-8aec-e3cf277e33fe`

## Logs:
- **db-cluster-98f6aa26.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/db-cluster-98f6aa26.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/db-cluster-98f6aa26.tar.gz)
- **monitor-set-98f6aa26.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/monitor-set-98f6aa26.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/monitor-set-98f6aa26.tar.gz)
- **loader-set-98f6aa26.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/loader-set-98f6aa26.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/loader-set-98f6aa26.tar.gz)
- **sct-runner-98f6aa26.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/sct-runner-98f6aa26.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/98f6aa26-f45b-4808-8aec-e3cf277e33fe/20220429_175546/sct-runner-98f6aa26.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-twcs-48h-test/30/)

**Comment by denesb** [comment]
Closing as stale.
